### ISSUE TITLE
The multigroup pincell xml example has memory errors

### ISSUE THREAD

**Issue by cjosey** [openmc-dev/openmc#1074]
As a side effect of asking a bunch of people to run the xml examples to get a handle of OpenMC, I keep getting informed of issues in the examples.

When examples/xml/pincell_multigroup is run, it will crash with the following error message:
```
*** Error in `../../../bin/openmc': free(): invalid next size (fast): 0x00000000011ba590 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x81429)[0x7fd1c2a97429]
/home/cjosey/codes/openmc/lib/libopenmc.so(+0x645e37)[0x7fd1c4c6de37]
/home/cjosey/codes/openmc/lib/libopenmc.so(__summary_MOD_write_summary+0x61)[0x7fd1c4c6e074]
/home/cjosey/codes/openmc/lib/libopenmc.so(__input_xml_MOD_read_input_xml+0x350)[0x7fd1c4bcf47e]
/home/cjosey/codes/openmc/lib/libopenmc.so(openmc_init_f+0x64)[0x7fd1c4ba19d1]
/home/cjosey/codes/openmc/lib/libopenmc.so(openmc_init+0x3c)[0x7fd1c4d1bbc4]
../../../bin/openmc(main+0x25)[0x400b27]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7fd1c2a383d5]
../../../bin/openmc[0x400a59]
======= Memory map: ========
00400000-00402000 r-xp 00000000 00:30 3753133                            /home/cjosey/codes/openmc/bin/openmc
00601000-00602000 r--p 00001000 00:30 3753133                            /home/cjosey/codes/openmc/bin/openmc
00602000-00603000 rw-p 00002000 00:30 3753133                            /home/cjosey/codes/openmc/bin/openmc
01165000-01378000 rw-p 00000000 00:00 0                                  [heap]
```
This was tested with GCC 8.1.0 and HDF5 1.10.1, compiled without alteration of ``CMakeLists.txt``.  As of v0.10.0, this issue did not occur.

**Comment by paulromano** [comment]
@nelsonag Can you take a look into this?

**Comment by nelsonag** [comment]
Will do

**Comment by paulromano** [comment]
Fixed by #1076.
